test/internal/layers/lazy.go

```diff
@@ -135,27 +130,85 @@ func (x *LazyImageLayers) extractLayers(ctx context.Context) (err error) {
 	return nil
 }
 
-func linuxImage(ctx context.Context, rc io.ReadCloser, dir string, _ []string) error {
-	f, err := os.Create(filepath.Join(dir, "layer.vhd"))
-	if err != nil {
-		return fmt.Errorf("create layer vhd: %w", err)
+func extractImageHandler(platform string, appendVerity bool) (extractHandler, error) {
+	var extract extractHandler
+	if platform == images.PlatformLinux {
+		extract = linuxExt4LayerExtractHandler()
+		if appendVerity {
+			extract = withAppendVerity(extract)
+		}
+		extract = withVhdFooter(extract)
```

Comment:
are we repeatedly writing to then reading and overwrititng the same file here?

Reply:
they'll be called only once, so we just append in the end, not recreating from scratch